ecosystem/sf-indexer/aptos-sf-indexer/migrations/2022-08-08-043603_core_tables/up.sql

```diff
@@ -0,0 +1,285 @@
+ "timestamp":"1649395495746947"
+ }
+ */
+CREATE TABLE block_metadata_transactions (
+  version BIGINT UNIQUE PRIMARY KEY NOT NULL,
+  block_height BIGINT NOT NULL,
+  id VARCHAR(255) NOT NULL,
+  round BIGINT NOT NULL,
+  epoch BIGINT NOT NULL,
+  previous_block_votes_bitvec jsonb NOT NULL,
+  proposer VARCHAR(255) NOT NULL,
+  failed_proposer_indices jsonb NOT NULL,
+  "timestamp" TIMESTAMP NOT NULL,
+  -- Default time columns
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  CONSTRAINT fk_versions FOREIGN KEY (version) REFERENCES transactions (version)
+);
+/* Ex:
+ {
+ "type":"user_transaction",
+ "version":"691595",
+ "hash":"0xefd4c865e00c240da0c426a37ceeda10d9b030d0e8a4fb4fb7ff452ad63401fb",
+ "state_root_hash":"0xebfe1eb7aa5321e7a7d741d927487163c34c821eaab60646ae0efd02b286c97c",
+ "event_root_hash":"0x414343554d554c41544f525f504c414345484f4c4445525f4841534800000000",
+ "gas_used":"43",
+ "success":true,
+ "vm_status":"Executed successfully",
+ "accumulator_root_hash":"0x97bfd5949d32f6c9a9efad93411924bfda658a8829de384d531ee73c2f740971",
+ "sender":"0xdfd557c68c6c12b8c65908b3d3c7b95d34bb12ae6eae5a43ee30aa67a4c12494",
+ "sequence_number":"21386",
+ "max_gas_amount":"1000",
+ "gas_unit_price":"1",
+ "expiration_timestamp_secs":"1649713172",
+ "payload":{
+ "type":"script_function_payload",
+ "function":"0x1::aptos_coin::mint",
+ "type_arguments":[
+ 
+ ],
+ "arguments":[
+ "0x45b44793724a5ecc6ad85fa60949d0824cfc7f61d6bd74490b13598379313142",
+ "20000"
+ ]
+ },
+ "signature":{
+ "type":"ed25519_signature",
+ "public_key":"0x14ff6646855dad4a2dab30db773cdd4b22d6f9e6813f3e50142adf4f3efcf9f8",
+ "signature":"0x70781112e78cc8b54b86805c016cef2478bccdef21b721542af0323276ab906c989172adffed5bf2f475f2ec3a5b284a0ac46a6aef0d79f0dbb6b85bfca0080a"
+ },
+ "events":[
+ {
+ "key":"0x040000000000000000000000000000000000000000000000000000000000000000000000fefefefe",
+ "sequence_number":"0",
+ "type":"0x1::Whatever::FakeEvent1",
+ "data":{
+ "amazing":"1"
+ }
+ },
+ {
+ "key":"0x040000000000000000000000000000000000000000000000000000000000000000000000fefefefe",
+ "sequence_number":"1",
+ "type":"0x1::Whatever::FakeEvent2",
+ "data":{
+ "amazing":"2"
+ }
+ }
+ ],
+ "timestamp":"1649713141723410"
+ }
+ */
+CREATE TABLE user_transactions (
+  version BIGINT UNIQUE PRIMARY KEY NOT NULL,
+  block_height BIGINT NOT NULL,
+  parent_signature_type VARCHAR(255) NOT NULL,
+  sender VARCHAR(255) NOT NULL,
+  sequence_number BIGINT NOT NULL,
+  max_gas_amount NUMERIC NOT NULL,
+  expiration_timestamp_secs TIMESTAMP NOT NULL,
+  gas_unit_price NUMERIC NOT NULL,
+  -- from UserTransaction
+  "timestamp" TIMESTAMP NOT NULL,
+  -- Default time columns
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  CONSTRAINT fk_versions FOREIGN KEY (version) REFERENCES transactions (version),
+  UNIQUE (sender, sequence_number)
+);
+CREATE INDEX ut_sender_index ON user_transactions (sender);
```

Comment:
this is redundant- PG automatically creates a unique index when a unique constraint or primary key is defined